### PR TITLE
Fix Combi-DRO

### PR DESCRIPTION
**Pull request by hansu**
This fixes two errors of the combi DRO:

1. Clicking on the DRO caused an exception
2. Calling of get_postion() failed, reported in #1398

### CODE REVIEW

File: lib/python/gladevcp/combi_dro.py [hunk 1]
```diff
@@ -629,13 +632,11 @@ def get_position(self):
...
               while the machine is moving, because of time delays
         '''
-        positions = self._position()
-        if self._ORDER == ["Rel", "Abs", "DTG"]:
-            return positions[1], positions[0], positions[2]
-        if self._ORDER == ["DTG", "Rel", "Abs"]:
-            return positions[2], positions[1], positions[0]
-        if self._ORDER == ["Abs", "DTG", "Rel"]:
-            return positions[0], positions[2], positions[1]
+        if self.diameter:
```
Comment (c-morley) [inline lib/python/gladevcp/combi_dro.py:635]: Does this scale the position no matter what axis?
I'm probably missing some details but only X axis should get scaled.
If this reinforced somewhere else?

Comment (hansu) [reply lib/python/gladevcp/combi_dro.py:635]: You have a Combi_DRO object for each axis. And `diameter` is a member of the Combi_DRO class, so you set it only for the X-axis object.
Here an example for gmoccapy ([go to file](https://github.com/LinuxCNC/linuxcnc/blob/2b07041d04501421cbd56a5fa0810e41beaa2214/src/emc/usr_intf/gmoccapy/gmoccapy.py#L1317-L1334)):
``` python
    def _make_lathe(self):
...
        # if we have a lathe, we will need an additional DRO to display
        # diameter and radius simultaneous, we will call that one Combi_DRO_9, as that value
        # should never be used due to the limit in axis from 0 to 8
        dro = Combi_DRO()
        dro.set_property("name", "Combi_DRO_9")
...
        joint = self._get_joint_from_joint_axis_dic("x")
        dro.set_joint_no(joint)
        dro.set_axis("x")
        dro.change_axisletter("D")
        dro.set_property("diameter", True)
...
```

File: lib/python/gladevcp/combi_dro.py [hunk 2]
```diff
@@ -424,18 +427,18 @@ def _position(self, object, p, rel_p, dtg, joint_actual_position):
         # joint_actual_position = joint positions, not needed here
         
         try:
-            dtg = dtg[self.axis_no]
-            abs_pos = p[self.axis_no]
-            rel_pos = rel_p[self.axis_no]
+            self.dtg = dtg[self.axis_no]
+            self.abs_pos = p[self.axis_no]
+            self.rel_pos = rel_p[self.axis_no]
```
Comment (hansu) [inline lib/python/gladevcp/combi_dro.py:432]: I can also take the scale factor at this position into account. Is maybe more logical here.

```suggestion
            dtg = dtg[self.axis_no]
            abs_pos = p[self.axis_no]
            rel_pos = rel_p[self.axis_no]
```            
```  
...
        if self.diameter:
            scale = 2.0
        else:
            scale = 1.0
...            
```
```diff
+           self.dtg = dtg * scale
+           self.abs_pos = abs_pos * scale
+           self.rel_pos = rel_pos * scale
```

Comment (hansu) [reply lib/python/gladevcp/combi_dro.py]: I have implemented this. Much better i think.

**Comment by c-morley** [comment]
Thank you

**Pull request merged**